### ISSUE TITLE
Images

### ISSUE THREAD

**Issue by tonyahowe** [LiteratureInContext/LiC-app#28]
The images that are available on github.com/tonyahowe/LiC are working as expected, but those in AWS S3 staging bucket are not--see franklin-autobiography.xml

**Comment by wsalesky** [comment]
Fixed. But we need to standardize how it is coded in the XML so I can change the image base root to stop using GitHub (see: https://github.com/LiteratureInContext/LiC-app/blob/development/repo-config.xml#L21)

Also Dave set up a nicer URL for us to use for the assets if you want the full URL's in the TEI:

> 
> We can serve via custom domains, say:
> 
>  
> 
> anthologyassets.lib.virginia.edu
> 
> anthologyassetsdev.lib.virginia.edu
> 
> (so, for example, your URL might be https://anthologyassetsdev.lib.virginia.edu/Haywood/images/HF-258.jpg and https://anthologyassets.lib.virginia.edu/Haywood/images/HF-258.jpg)

Let me know if you have any questions.

**Comment by tonyahowe** [comment]
Thanks, Winona! We'll work on that when we return!

**Comment by jobrien62** [comment]
OK, good; thanks for fixing things for now. John

Sent from my iPad

On Mar 21, 2019, at 4:47 PM, Tonya Howe <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks, Winona! We'll work on that when we return!

**Comment by brinascode** [comment]
Hi Winona,
We've worked on standardizing the picture urls and so far they look like this:
https://s3.amazonaws.com/lic-assets-staging/Shelley/vol1/0.01.jpg . 
We have an idea of how to organize the pictures on S3 and soon I'll write a reference for the naming convention. 
I also used the custom domain for some pictures and it works great. 
Could you please program the app so that whenever it finds facs=" it appends either
1) https://s3.amazonaws.com/lic-assets-staging or the custom domains from your last comment? 
That would be great! 
Thanks

**Comment by wsalesky** [comment]
@brinascode I have made this change. Let me know if you see any issues.

**Comment by brinascode** [comment]
It works thank you!

**Comment by brinascode** [comment]
I am getting this error for a few images, even though in the XML my link is facs="Shelley/vol1/0.0001.jpg" an not ="/Shelley...

<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>
/Shelley/http://anthologyassetsdev.lib.virginia.edu/Shelley/vol1/0.001.jpg
</Key>
<RequestId>FDE88003407EF866</RequestId>
<HostId>
5S2qUrSvusOBnSg4x3YZBR7/6hDkJd/gvwGo+wHNB0jHdh0dVNwvhMGo+cu6WT+FurbmY8WOT64=
</HostId>
</Error>

**Comment by wsalesky** [comment]
@brinascode What is the image path convention that will be used in the XML? There are a few different ways of linking to page images in this file: https://anthologydev.lib.virginia.edu/work/Shelley/shelley-frankenstein-1818.tei

example: 
1. `<pb n="[title page]" facs="Shelley/vol1/0.0001.jpg"/>`

2. `<pb n="[title page verso]" facs="http://anthologyassetsdev.lib.virginia.edu/Shelley/vol1/0.001.jpg"/>`

3. `<pb n="[dedication]" facs="https://s3.amazonaws.com/lic-assets-staging/Shelley/vol1/0.01.jpg"/>`

4. `<pb n="[1.vii]" facs="images/vii.jpg"/>`

If they are not going to include the `http://anthologyassetsdev.lib.virginia.edu/` then they will need to follow a common convention, so I can the rest of the path correctly. I think the current script tries to add the work directory (in this case '/Shelley/'), this works with example 4 but not with example 1. I can code around it, but it would be better to just be standardized in the TEI. So let me know which to expect and I will make that work. 

Thanks!

**Comment by brinascode** [comment]
The first one is what we would like to stick with; I have yet to fix the other links.

**Comment by wsalesky** [comment]
Okay. I have fixed the code. Let me know if it does not behave as expected.

**Comment by brinascode** [comment]
It works awesome thanks!

**Comment by wsalesky** [comment]
Great. We should probably build some sort of paging mechanism to page through the text and images, as loading all those images on a single page will really slow things down. Just to keep in mind.

**Comment by brinascode** [comment]
True, that would be a good idea

**Comment by tonyahowe** [comment]
Winona, we continue to work with the images, and are hoping you can implement something that will make the process a little more automated and less prone to typos!

Each xml file has a unique xml:id, like behn-rover. Can we use these xml:ids as the main folder for page and note images related to it? For instance, if we store the images on AWS as [xml:id]/pageImages/xxx.jpg and [xml:id]/notes/xxx.jpg, where [xml:id] is, for instance, behn-rover, can we simplify another level of the process? 

Then, in the xml, we would reference facs="pageImages/xxx.jpg" or url="notes/xxx.jpg", with the xml:id being filled in from the xml file itself?

**Comment by wsalesky** [comment]
Yes, as long as the pattern is consistent and the xml:id matches the folder on AWS that is fine. Let me know what the final pattern is.

**Comment by tonyahowe** [comment]
Okay-- using finch-nightingale.xml as an example, I added pages images as <pb facs="pageImages/xxx.jpg"/> and any images as <graphic url="notes/xxx.jpg"/>. There are corresponding images on AWS, under the folder finch-nightingale. 

If this works, it's also a good way to keep things clear around multiple texts by the same author!

**Comment by wsalesky** [comment]
Done.

**Comment by brinascode** [comment]
Hi Winona,

It seems like the images inside graphic tags arent't being linked to AWS.
For example, in the xml there is:
 <graphic url="notes/Benjamin_Franklin_by_Joseph_Siffrein_Duplessis.jpg" style="float:right" width="400px"/>

but in the app the picture does not show up, and when we click on the image address of the little image icon it gives: https://anthologydev.lib.virginia.edu/work/Franklin/notes/Jonathan-Shipley.jpg
instead of using the xml:id.

Thanks!

**Comment by wsalesky** [comment]
@brinascode 

Try now please. The code will use the supplied url if it is a full url (starting with 'http' or 'https'), if it is a relative url it will add the work id and the AWS path. 

Close if it is resolved.

**Comment by brinascode** [comment]
Awesome thanks it works.
Closing the issue.

**Comment by tonyahowe** [comment]
pageImages/vol1/0001.jpg --> books with multiple volumes
pageImages/0001.jpg --> no volumes
notes/xxxxx.jpg --> all images linked in notes from AWS
or a live url with a stable image file
can we style the images: float:right width:300px

**Comment by wsalesky** [comment]
@tonyahowe Can I get an example XML to test the variouse image permutations? One XML example for each image format listed above. 
Thanks!
-Winona

**Comment by tonyahowe** [comment]
AWS//xml:id/pageImages/vol1/xxx.jpg --> books with multiple volumes --> **shelley-frankenstein-1818.xml**
AWS//xml:id/pageImages/0001.jpg --> no volumes to worry about --> most all others
AWS//xml:id/notes/xxxx.jpg --> all images linked in notes from AWS --> **franklin-autobiography.xml**
OR: a live url with a stable image file. -->

Can we style all **NOTES** images with the following CSS (or thereabouts): float:right width:300px --> see examples here:

- from franklin-autobiography.xml : ` <note xml:id="n0002" target="a0002" type="editorial" resp="editors.xml#JOB">Jonathan Shipley (1714-1788) <p><graphic url="notes/Jonathan-Shipley.jpg" style="float:right" width="300px"/>` 
- from haywood-fantomina.xml: `<docAuthor>By <ref target="a002" corresp="n002">Mrs. ELIZA HAYWOOD.</ref>

               <note xml:id="n002" target="a002" type="editorial" resp="editors.xml#TH">
                  <graphic style="float:right" url="https://upload.wikimedia.org/wikipedia/commons/2/25/Eliza-haywood.jpg"/>
                  Eliza Haywood (c.1693-1756) was a prolific author, actor, and publisher of the
                  early- to mid-eighteenth century. She is most famous, today, for her novels and
                  novellas, among which <hi rend="italic">Fantomina</hi>is numbered. The image
                  included here, <ref
                     target="https://commons.wikimedia.org/wiki/File:Eliza-haywood.jpg">via
                     Wikimedia Commons</ref>, is an engraved frontispiece portrait by George Vertue.
                  Haywood wrote in a number of different genres, including amatory fiction, domestic
                  fiction, and essay.</note><lb/></docAuthor>`

**Basically, we don't want students to have to insert the styling. I will delete all the float and width attributes on haywood-fantomina.xml as a test case!**

Can the images expand if clicked? Go to the source url if clicked? Not necessary, but if it's easy, that would be nice.

- In the case of the AWS//notes/image.jpg cases (as in franklin-autobiography.xml), these should show the full-size image on AWS
- in the case of the (hopefully) stable URLs (as in haywood-fantomina.xml), these should redirect to the URL online

**Comment by tonyahowe** [comment]
It seems that the images in footnotes aren't visible now--see note in #44

**Comment by wsalesky** [comment]
@tonyahowe Sorry that is a bug introduced by the new page image view. I will get it fixed Monday

**Comment by jobrien62** [comment]
Working now, looks like!

**Comment by tonyahowe** [comment]
Hi, Winona! I went through and put the images from notes into AWS, as several links I'd thought of as stable (British Museum, etc) had moved. They are showing up when viewed from the main source page, but the embedded images are not showing up when viewed through the contributors page (see SS below).
<img width="799" alt="Screen Shot 2019-06-12 at 1 46 48 PM" src="https://user-images.githubusercontent.com/1153464/59373976-ca0b9900-8d18-11e9-8426-96e6f14f7d27.png">

**Comment by tonyahowe** [comment]
Hi, @wsalesky! Can we wrap the images in the contributors' display the same way we did in the popups? This is what the contributors' display looks like now:
<img width="844" alt="Screen Shot 2019-06-12 at 2 55 25 PM" src="https://user-images.githubusercontent.com/1153464/59378348-1c04ec80-8d22-11e9-95d9-ed9f7f1d5fcb.png">
